### PR TITLE
Add full param options for monitors

### PR DESCRIPTION
**Pull request by Seanstoppable**
Add a more generic function that will let people mix and match
Have existing functions wrap this better function
Resolves the rest of state support in #164

### CODE REVIEW

**Comment by Seanstoppable** [comment]
Great, thanks @ojongerius

File: monitors.go [hunk 1]
```diff
@@ -171,11 +151,59 @@ func (client *Client) DeleteMonitor(id int) error {
...
+		value := fmt.Sprintf("group_states=%v", strings.Join(opts.GroupStates, ","))
+		query = append(query, value)
+	}
+
+	if len(opts.MonitorTags) > 0 {
+		value := fmt.Sprintf("monitor_tags=%v", strings.Join(opts.MonitorTags, ","))
+		query = append(query, value)
+	}
+
+	if opts.WithDowntimes != nil {
```
Comment (ojongerius) [inline monitors.go]: Up to you but you could use the generated GetWithDownTimesOk like so:

```go
if v, ok := opts.GetWithDowntimesOk; ok {
    query = append(query, fmt.Sprintf("name=%t", strings.Join(v, ",")))
}
```

File: monitors.go [hunk 2]
```diff
@@ -171,11 +151,59 @@ func (client *Client) DeleteMonitor(id int) error {
...
+		value := fmt.Sprintf("with_downtimes=%t", *opts.WithDowntimes)
+		query = append(query, value)
+	}
+
+	if opts.Name != "" {
```
Comment (ojongerius) [inline monitors.go]: Works at the moment but needs to be changed either manually test if not nil if you refactor Name to be a pointer. Alternatively you can use the autogenereated getters and setters like so:

```go
if v, ok := opts.GetNameOk; ok {
    query = append(query, fmt.Sprintf("name=%v", strings.Join(v, ",")))
}
```

**Comment by Seanstoppable** [comment]
@ojongerius apologies for the delay. Updated with feedback, thanks!

**Comment by ojongerius** [comment]
@Seanstoppable sorry I've been swamped with work, could you please resolve conflicts so I can merge this?

**Comment by Seanstoppable** [comment]
Closing in favor of #199
